compiler/rustc_lint/src/unused.rs

```diff
@@ -1053,17 +1053,31 @@ impl UnusedParens {
             self.emit_unused_delims(cx, value.span, spans, "pattern", keep_space, false);
         }
     }
+
+    fn cast_followed_by_lt(&self, e: &ast::Expr) -> Option<ast::NodeId> {
+        if let ExprKind::Binary(op, lhs, _rhs) = &e.kind
+            && (op.node == ast::BinOpKind::Lt || op.node == ast::BinOpKind::Shl)
+        {
+            let rhs_of_lhs = match &lhs.kind {
+                ExprKind::Binary(_, _, rhs) => rhs,
```

Comment:
Ideally, this should probably be a recursion, to cover cases like `a + b * c as (u8) < 10` and further.

Reply:
yeah, fixed and add a test case.